### ISSUE TITLE
Use VideoProgressListener?

### ISSUE THREAD

**Issue by sheriffmarley** [pascalbaljetmedia/laravel-ffmpeg#71]
I need to have the progress of any currently converting video. I found the progress method but there you only receive the current percentage.

In the `phpffmpeg` package I found a class called `VideoProgressListener` which extends the `AbstractProgressListener`. So how can I use the `VideoProgressListener`? If I see it correctly there are a bunch of values like `remaing`, `rate`,`percentage` etc., exactly what I need.
How am I supposed to use the `VideoProgressListener` with laravel? 

I tried to create a new Instance in the progress event, but that seems to be the wrong approach.

**Comment by pascalbaljet** [comment]
Will look into it, maybe it’s for internal use only (in the underlying package)?

**Comment by pascalbaljet** [comment]
It's been a while, but I figured the only way to get access to the `ProgressListener` is to decorate the `Format` instance. Would something like this work for you?

```php
<?php

$format = new X264;
$decoratedFormat = ProgressListenerDecorator::decorate($format);

$decoratedFormat->on('progress', function ($video, $format, $percentage) use ($decoratedFormat) {
    $listeners = $decoratedFormat->getListeners();
    $listeners[0]->getCurrentTime();
});

(new MediaOpener)
    ->open('video.mp4')
    ->export()
    ->inFormat($decoratedFormat)
    ->save('new_video.mp4');
```

**Comment by pascalbaljet** [comment]
Fixed in `v7.0`
